### ISSUE TITLE
Pulling singularity images vs. docker images

### ISSUE THREAD

**Issue by civier** [NeuroDesk/vnm#83]
Is it really faster for international users to pull docker images over singularity images?
If there was a singularity hub comparable to docker hub (I don't really know the current state of things), wouldn't it be faster to pull singularity images diretly?

On one hand, the singularity images are usually smaller than the docker images + we save the time required to unpack and convert to sif (which is quite long, unfortunately).
On the other hand, docker images that share layers will obviously be downloaded faster.

@stebo85 @aswinnarayanan  Do you know if the packages share a lot of stuff between them? how big are the base layers they share?

**Comment by stebo85** [comment]
yes, I think storing singularity images would be nice, but currently none of the available singularity registries works well enough :)

**Comment by civier** [comment]
So how much data can we actually store on Docker hub? And does it have caps on the total GBs downloaded from our repository per day/week/month? Can we buy more storage/throughput?

**Comment by stebo85** [comment]
There are no limits on docker hub or github - however there are limits on singularity hub - that's why we use docker for now :)

**Comment by civier** [comment]
So who pays for the docker hub servers?
Is it financed by docker? How long should we assume it will stay
free-of-charge for?

Oren

**Comment by stebo85** [comment]
yes, docker pays for dockerhub and microsoft pays for github. Since we have our containers in these 2 registries we can be pretty sure that not both go away at the same time :)

**Comment by aswinnarayanan** [comment]
@civier 
Github Packages is free for Public Repos, no limits currently -- runs on MS azure infrastructure.
> https://github.com/features/packages#pricing

Github packages doesn't even let us delete packages from the public repos in order to not break downstream dependacies , so they're actually encouraging us to keep the images available indefinitely.
> https://docs.github.com/en/packages/publishing-and-managing-packages/deleting-a-package#about-public-package-deletion

I agree with @stebo85. If dockerhub were to restrict capacities for free repos, it would have major far reaching consequences...
For instance, the images our builds are pulling from would also be affected.... we are very downstream in comparison

**Comment by stebo85** [comment]
I built a singularity caching mechanisms that keeps all images we currently use in neurodesk on swift storage. The primary location of containers will be dockerhub, but I am hoping that caching the images as singularity files will speed up the installation process:

What do people think about this?
https://github.com/NeuroDesk/neurodesk/blob/master/release_new_version.sh
